src/main/java/uk/gov/hmcts/ccd/domain/service/getevents/DefaultGetEventsOperation.java

```diff
@@ -58,4 +60,18 @@ public DefaultGetEventsOperation(CaseAuditEventRepository auditEventRepository,
         return auditEventRepository.findByEventId(eventId).map(Optional::of)
             .orElseThrow(() -> new ResourceNotFoundException(CASE_EVENT_NOT_FOUND));
     }
+
+    @Override
+    public List<AuditEvent> getEvents(String caseReference) {
+        if (!uidService.validateUID(caseReference)) {
+            throw new BadRequestException("Case reference " + caseReference + " is not valid");
+        }
+
+        final CaseDetails caseDetails =
+            getCaseOperation.execute(caseReference)
+                .orElseThrow(() -> new ResourceNotFoundException(
+                    String.format(CASE_RESOURCE_NOT_FOUND, caseReference)));
+
+        return getEvents(caseDetails);
+    }
```

Comment:
this is duplicated with the already existing `getEvents(String jurisdiction, String caseTypeId, String caseReference)
The only difference between the 2 is that the above creates a different error message which indicates the  jurisdiction and caseTypeId.
```String.format(RESOURCE_NOT_FOUND, jurisdiction, caseTypeId, caseReference)));```
 I think that's not an important detail and can be changed to only log the caseReference. If you do that you can find a way to easily reuse that logic rather than duplicating

Reply:
Did some changes.... Please review the code